dbcon/joblist/tupleaggregatestep.cpp

```diff
@@ -5688,14 +5688,16 @@ uint64_t TupleAggregateStep::doThreadedAggregate(ByteStream& bs, RowGroupDL* dlp
       // much memory on average
       uint32_t threads = std::max(1U, fNumOfThreads / 2);
       runners.reserve(threads);
-      for (i = 0; i < threads; ++i)
+      for (uint32_t threadNum = 0; threadNum < threads; ++threadNum)
       {
-        runners.push_back(jobstepThreadPool.invoke(ThreadedAggregateFinalizer(this, i)));
+        runners.push_back(jobstepThreadPool.invoke(ThreadedAggregateFinalizer(this, threadNum)));
       }
       jobstepThreadPool.join(runners);
     }
 
-    if (dynamic_cast<RowAggregationDistinct*>(fAggregator.get()) && fAggregator->aggMapKeyLength() > 0)
+    auto* distinctAgg = dynamic_cast<RowAggregationDistinct*>(fAggregator.get());
```

Comment:
IMHO it is worth to rename this into aggFuncWithDistinct. What do you think?

Reply:
you mean renaming `distinctAgg`? Totally agree, would make the semantics more clear.

Reply:
I went with `distinctAggregator` for the Aggregator that accumulates the results over all row groups and renamed the bucketAggregators accordingly to allow for easier distinguishment.